src/pages/cart-page/CartPage.jsx

```diff
@@ -0,0 +1,74 @@
+import "./cart-page.css";
+import {
+  CardHorizontal,
+  CartPrice,
+  Footer,
+  Navigation,
+} from "../../components";
+import { useCartContext, useWishlistContext } from "../../context";
+import { addToWishlist } from "../../utility";
+
+const CartPage = () => {
+  // wishlist Context
+  const { wishlists, setWishlists } = useWishlistContext();
+
+  // cart context
+  const {
+    cartState: { cartList },
+    dispatch,
+  } = useCartContext();
+
+  // cart price section calculations
+  const price = cartList.reduce(
+    (acc, item) => acc + Number(item.originalPrice),
+    0
+  );
+  const discountPrice = cartList.reduce(
+    (acc, item) =>
+      acc + (Number(item.originalPrice) - Number(item.discountPrice)),
+    0
+  );
+  const totalPrice = price - discountPrice;
+
+  return (
+    <>
+      <Navigation />
+      <main>
+        <h2 className="text-center">MY CART</h2>
+        <div className="cart-page-wrapper">
+          <section className="cart-product">
+            {cartList.length > 0 &&
+              cartList.map((item) => (
```

Comment:
You can try cartList?.map than you don't have to use this ternary opertaor to check whether list is empty or not

Reply:
i think optional chaining doesn't work for an empty array, as the initial state for cartList is [ ]